everyvoice/wizard/dataset.py

```diff
@@ -689,20 +727,198 @@ def effect(self):
             self.state["filelist_headers"]
         )
         # re-parse data:
-        reload_filelist_data_as_dict(self.state)
+        convert_filelist_data_to_dict(self.state)
         # Add speaker IDs if they are not specified in the filelist
         if self.state[StepNames.data_has_speaker_value_step] == "no":
             add_missing_speaker(self.state)
         # Apply the language code:
-        isocode = get_iso_code(self.response)
-        # Apply text conversions and get target training representation
-        self.state["model_target_training_text_representation"] = (
-            apply_automatic_text_conversions(
-                self.state["filelist_data"],
-                self.state[StepNames.filelist_text_representation_step],
-                global_isocode=isocode,
+        for item in self.state["filelist_data"]:
+            item["language"] = self.response
+
+
+class CustomG2PStep(Step):
+    DEFAULT_NAME = StepNames.custom_g2p_step
+    REVERSIBLE = True
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.found_languages = False
+
+    def find_languages(self):
+        """Enumerate the languages this dataset uses."""
+        from g2p import get_arpabet_langs
+
+        from everyvoice.text.phonemizer import (
+            AVAILABLE_G2P_ENGINES,
+            DEFAULT_G2P,
+            CachingG2PEngine,
+        )
+
+        self.language_codes = sorted(
+            set(row["language"] for row in self.state["filelist_data"])
+        )
+
+        _, g2p_langs_full = get_arpabet_langs()
+        self.language_names = [
+            g2p_langs_full.get(language, language) for language in self.language_codes
+        ]
+
+        self.current_engines = []
+        self.set_a_custom = []
+        for language in self.language_codes:
```

Comment:
What if someone labelled two different datasets as `crk` and then realized that actually they should have labelled them `crk-east` and `crk-west` since they use different g2p engines - is there a way to edit that at this point? Or would the recommendation be to edit the config file?